app/javascript/shared/components/UserStore.js

```diff
@@ -0,0 +1,60 @@
+export class UserStore {
+  constructor() {
+    this.users = [];
+    this.wasFetched = false;
+  }
+
+  fetch(url) {
+    const myStore = this;
+    return new Promise((resolve, _reject) => {
+      if (myStore.wasFetched) {
+        resolve();
+      } else {
+        window
+          .fetch(url)
+          .then((res) => res.json())
+          .then((data) => {
+            myStore.users = data.reduce((array, aUser) => {
+              array.push(aUser);
+              return array;
+            }, []);
+            myStore.wasFetched = true;
+            resolve();
+          })
+          .catch((error) => {
+            // console.log("what error?", error)
```

Comment:
just nodding to this for now, I see things are still WIP per the PR description so assuming it's for debugging